packages/Kokkos/src/DTK_LinearBVH.cpp

```diff
@@ -0,0 +1,69 @@
+#include <details/DTK_DetailsAlgorithms.hpp>
+#include <details/DTK_DetailsTreeConstruction.hpp>
+#include <details/DTK_DetailsTreeTraversal.hpp>
+
+#include <DTK_LinearBVH.hpp>
+
+#include <Kokkos_ArithTraits.hpp>
+
+namespace DataTransferKit
+{
+
+BVH::BVH( AABB const *bounding_boxes, int n )
+    : _leaf_nodes( "leaf_nodes", n )
+    , _internal_nodes( "internal_nodes", n - 1 )
+    , _indices( "sorted_indices", n )
+{
+    // determine the bounding box of the scene
+    Details::calculateBoundingBoxOfTheScene( bounding_boxes, n,
+                                             _internal_nodes[0].bounding_box );
+    // calculate morton code of all objects
+    Kokkos::View<unsigned int *, DeviceType> morton_indices( "morton", n );
+    for ( int i = 0; i < n; ++i ) // parallel_for
+        morton_indices[i] = Kokkos::ArithTraits<unsigned int>::max();
+    Details::assignMortonCodes( bounding_boxes, morton_indices.data(), n,
+                                _internal_nodes[0].bounding_box );
```

Comment:
I don't quite agree with you here.  The first internal node is the root and its bounding box will be the bounding box of the scene/global bounding box.